controller.py

```diff
@@ -0,0 +1,79 @@
+from pid import PID
+from rocket import Rocket, GRAVITY
+
+
+class Controller:
+	def __init__(self, state='hover', rocket=None, pid=None):
+		self.state = state
+
+		# Build a rocket!
+		if rocket is not None:
+			self.rocket = rocket
+		else:
+			self.rocket = Rocket(
+				dry_mass = 100,
+				fuel = 800,
+				altitude=1,
+				velocity=0,
+				fuel_consumption=1,
+				thrust = 9000
+			)
```

Comment:
Why is the controller creating the rocket instead of otherwise? It seems more intuitive that you'll be creating a rocket first and it will have its own controller. And also, initializing a rocket needs 6 parameters, and since it is initialized in the controller, you would need to change the physics related parameters for a rocket inside its controller or have the data juggled around in the whole engine, which is just overcomplication and unnecessary.